### PR TITLE
TC-180 uk tariff & TC-178 refreshing commodity tree

### PR DESCRIPTION
**Pull request by andrzejlichaczewski**

### CODE REVIEW

File: dit_helpdesk/hierarchy/models.py [hunk 1]
```diff
@@ -15,31 +15,84 @@
...
+
+    def __init__(self):
+        pass
+
+    @classmethod
+    def get_active_objects(cls, region):
+        return cls.all_objects.filter(
+            nomenclature_tree__region=region,
+            nomenclature_tree__end_date__isnull=True,
+        )
```
Comment (kevincarrogan) [inline dit_helpdesk/hierarchy/models.py:36]: Could this be better as a manager method or is this easier to be attached to the models?

File: dit_helpdesk/hierarchy/models.py [hunk 2]
```diff
@@ -15,31 +15,84 @@
...
+
+    def save(self):
+        if not self.nomenclature_tree:
+            self.nomenclature_tree = NomenclatureTree.get_active_tree('EU')
```
Comment (kevincarrogan) [inline dit_helpdesk/hierarchy/models.py:40]: `EU` is set as the default argument for `get_active_tree` so is there a reason we want to be explicit here?

File: dit_helpdesk/hierarchy/models.py [hunk 3]
```diff
@@ -190,10 +243,12 @@ def get_ancestor_data(self, parent=None, tree=None, level=0):
         return tree
 
 
-class Chapter(models.Model):
+class Chapter(models.Model, TreeSelectorMixin):
     """
     Model representing the second level chapters of the hierarchy
     """
+    objects = EUHierarchyManager()
+    all_objects = models.Manager()
```
Comment (kevincarrogan) [inline dit_helpdesk/hierarchy/models.py:251]: We appear to be doing this a lot, I'm wondering if we move some of this up into a base class for each of these models to inherit from.

File: dit_helpdesk/hierarchy/models.py (hunk 2 above)
Comment (andrzejlichaczewski) [reply dit_helpdesk/hierarchy/models.py:40]: Just to kind of remind ourselves that by default almost everything applies only to the `EU` tree.

File: dit_helpdesk/hierarchy/models.py (hunk 3 above)
Comment (andrzejlichaczewski) [reply dit_helpdesk/hierarchy/models.py:251]: We could do that, it's 2 lines x 5 models though so it didn't feel that painful to me yet.. it is something that repeats though so can be abstracted a bit.

File: dit_helpdesk/hierarchy/models.py (hunk 2 above)
Comment (andrzejlichaczewski) [reply dit_helpdesk/hierarchy/models.py:40]: It does mean though that if we change we'd have to change it to `UK` here as well as in the default parameter definition.

File: dit_helpdesk/hierarchy/models.py (hunk 1 above)
Comment (andrzejlichaczewski) [reply dit_helpdesk/hierarchy/models.py:36]: Yeah I think that's an option.. so something like `Commodity.all_objects.get_active('UK')`

**Review by kevincarrogan — APPROVED** [review]

**Pull request merged**